### PR TITLE
test that rewards are smaller for smaller stake

### PR DESCRIPTION
**Pull request by piotr-iohk**
# Issue Number

#1720

# Overview

- 3fbdeed9b06d46619dcdbf685cae98cd16711381
  test that rewards are smaller for smaller stake
  
- e94100736e35a1a98e75d39c66d161690d5a4049
  compare sum of rewards from pools for small and big stake

# Comments

### CODE REVIEW

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs [hunk 1]
```diff
@@ -606,14 +606,27 @@ spec = do
...
+            eventually "rewards are smaller for smaller stakes" $ do
+                let stakeSmall = Just (Coin 10_000_000)
+                let stakeBig = Just (Coin 1000_000_000)
+                Right poolsStakeSmall <- snd <$> listPools ctx stakeSmall
+                Right poolsStakeBig <- snd <$> listPools ctx stakeBig
+                let rewards = view (#metrics . #nonMyopicMemberRewards)
+                let rewardsStakeSmall = rewards <$> poolsStakeSmall
+                let rewardsStakeBig = rewards <$> poolsStakeBig
+
+                rewardsStakeSmall `shouldSatisfy` (< rewardsStakeBig)
```
Comment (Anviking) [inline lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs]: I think our custom `.<` will have a yield nicer failure reasons than `shouldSatisfy`

Comment (Anviking) [reply lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs]: You're relying on `Ord [a]`. I'm not sure exactly how that works
```haskell
λ> [3,4] > [3,4]
False
λ> [3,5] > [3,4]
True
λ> [2,5] > [3,4]
False
```
🤔 

Maybe we are looking for two checks
- For one or more pools, higher stake results in _strictly_ higher member rewards
- For all pools, higher stake results in _higher or the same_ member rewards (could imagine them being zero)
?

Or perhaps that is an unnecessarily complex check. Also, the ordering of pools _could_ change between the requests (on rare occasions)

Maybe just comparing the `sum` of the rewards would be more resilient, simple and enough?

**Review by Anviking — APPROVED** [review]
Nice!

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs (hunk 1 above)
Comment (piotr-iohk) [reply lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs]: okay, I went for comapring sums of rewards. :sweat_smile:

**Pull request merged**